### PR TITLE
Allows traders to strengthen their armor by adding goliath hides just like miners.

### LINKED ISSUE #12928: Vox trader armor shouldn't be near-useless
Vox trader armor used to have very good protection (which was bit too much) but now it is complete garbage with no defense, no flash block, and they can't hold a jetpack like they used to. Traders shouldn't be walking around with essentially no protection considering their outpost is on the asteroid where there are plenty of things ready to fuck your day up and a trade gone wrong can end up in death. What should be changed is that they revert to how they used to work and their armor values lowered to what mining hardsuits have.

### PR DESCRIPTION
**Pull request by Probe1**
Closes https://github.com/d3athrow/vgstation13/issues/12928

🆑 

 * rscadd: Vox Traders can now use goliath hides to fortify their suits and helmets.

### CODE REVIEW

**Comment by ComicIronic** [comment]
>typelist

Disgusting.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 1]
```diff
@@ -497,7 +497,7 @@ obj/item/asteroid/basilisk_hide/New()
 
 /obj/item/asteroid/goliath_hide/afterattack(atom/target, mob/user, proximity_flag)
 	if(proximity_flag)
-		if(istype(target, /obj/item/clothing/suit/space/rig/mining) || istype(target, /obj/item/clothing/head/helmet/space/rig/mining) || istype(target, /obj/item/clothing/suit/space/plasmaman/miner) || istype(target, /obj/item/clothing/head/helmet/space/plasmaman/miner))
+		if(istype(target, /obj/item/clothing/suit/space/rig/mining) || istype(target, /obj/item/clothing/head/helmet/space/rig/mining) || istype(target, /obj/item/clothing/suit/space/plasmaman/miner) || istype(target, /obj/item/clothing/head/helmet/space/plasmaman/miner) || istype(target, /obj/item/clothing/suit/space/vox/civ/trader) || istype(target, /obj/item/clothing/head/helmet/space/vox/civ/trader))
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: We seriously need an is_goli_reinforcable() check or something, this is getting a tad ridiculous

Comment (Probe1) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: ``` if(istype(WYCI, /obj/module/work_in_progress/things_you_care_about)) ```

``` else do_real_work ```

Comment (PJB3005) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Nope.

Code quality.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Okay, gimme a minute

**Comment by Probe1** [comment]
@PJB3005 I'm not doing shit until you clearly explain what is wrong then cite where else in the code it is handled how you would like it to be. 

I'm not dealing with "I just invented a style requirement".

**Comment by Intigracy** [comment]
He/they mean a clothing proc that returns false by default but returns true for things that can be reinforced via goliath hides.

Istype lists are horrid for maintainability.

**Review by PJB3005 — APPROVED** [review]

**Comment by Intigracy** [comment]
sure why not

**Pull request merged**